### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by MetRonnie**
This is a small change with no associated Issue.

- Fix bug where installing a `suite.rc` workflow that had been `cylc validate`d in its source dir would result in a `flow.cylc` symlink being copied over still pointing to the source dir `suite.rc` instead of the one in the run dir
- Fix bug where the deprecated `flow.cylc` config item

  ```ini
  [scheduling]
      [[dependencies]]
          graph = foo
  ```
  (skipped `[[[<recurrence>]]]` section) did not log a warning when upgraded to

  ```ini
  [scheduling]
      [[graph]]
          graph = foo  # (same as R1 = foo)
  ```
- Do not show traceback when any `CylcError` gets raised during running of the workflow, only show traceback for generic uncaught exceptions (unless using `--debug`, in which case show traceback for all exceptions)

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Appropriate tests are included (unit and/or functional).
- [x] Appropriate change log entry included.
- [x] No documentation update required.
- [x] No dependency changes.

### CODE REVIEW

File: cylc/flow/rundb.py [hunk 1]
```diff
@@ -167,7 +167,7 @@ class CylcSuiteDAO:
     CONN_TIMEOUT = 0.2
     DB_FILE_BASE_NAME = "db"
     MAX_TRIES = 100
-    RESTART_INCOMPAT_VERSION = "8.0a2"  # Can't restart suite if <= this vers
+    RESTART_INCOMPAT_VERSION = "8.0b0"  # Can't restart suite if <= this vers
```
Comment (MetRonnie) [inline cylc/flow/rundb.py:170]: 8.0a2 was never released

File: cylc/flow/scheduler.py [hunk 2]
```diff
@@ -616,7 +614,6 @@ async def run(self):
 
         """
         try:
-            await self.install()
```
Comment (MetRonnie) [inline cylc/flow/scheduler.py:619]: `Scheduler.install()` has already just been run in `scheduler_cli._setup()` https://github.com/cylc/cylc-flow/blob/dd8e26ac9cbba3c4b4683380456e59c52e000d8d/cylc/flow/scheduler_cli.py#L293-L310

https://github.com/cylc/cylc-flow/blob/dd8e26ac9cbba3c4b4683380456e59c52e000d8d/cylc/flow/scheduler_cli.py#L334-L337

https://github.com/cylc/cylc-flow/blob/dd8e26ac9cbba3c4b4683380456e59c52e000d8d/cylc/flow/scheduler_cli.py#L342-L347

**Comment by MetRonnie** [comment]
Fast test failures seem to be due to https://github.com/pytest-dev/pytest-asyncio/issues/209

File: cylc/flow/scheduler.py (hunk 2 above)
Comment (oliver-sanders) [reply cylc/flow/scheduler.py:619]: Hmmm, why the heck was it like that.

Would it work to keep the install in the Scheduler and remove it from `scheduler_cli`?

**Comment by oliver-sanders** [comment]
> passing locally for me on pytest-asyncio 0.14.0

Passing locally for me on pytest-asyncio 0.15.0 but failing on GH actions?!

I've pinned us back to 0.14.0 on #4176 until a fix is out

File: cylc/flow/scheduler.py (hunk 2 above)
Comment (MetRonnie) [reply cylc/flow/scheduler.py:619]: I tried removing `scheduler_cli._setup()` first, but the
```python
 # setup loggers 
 _open_logs(reg, options.no_detach) 
```
bit fails without the workflow run dir tree set up

Comment (oliver-sanders) [reply cylc/flow/scheduler.py:619]: Dammit, nvm.

**Comment by MetRonnie** [comment]
Final test failure should be fixed by #4185

**Review by oliver-sanders — APPROVED** [review]

**Comment by hjoliver** [comment]
(Cancelled Actions after merging a trivial CHANGES.md conflict)

**Review by hjoliver — APPROVED** [review]
Good :+1:

**Pull request merged**